R/tag_query.R

```diff
@@ -1087,40 +1092,40 @@ tagQueryAttrsRemove <- function(els, attrs) {
   }
   tagQueryWalk(els, function(el) {
     if (!isTagEnv(el)) return()
-    for (attrVal in attrs) {
-      el$attribs[[attrVal]] <- NULL
-    }
-  })
-}
-# Remove attribute values
-tagQueryAttrsEmpty <- function(els) {
-  tagQueryWalk(els, function(el) {
-    if (!isTagEnv(el)) return()
-    el$attribs <- list()
+    tagEnvRemoveAttribs(el, attrs)
   })
 }
 # Check if els have attributes
-tagQueryAttrHas <- function(els, attr) {
-  attr <- as_character2(attr)[[1]]
-  if (length(attr) != 1 || !is.character(attr)) {
-    stop("`attr` must be a single character value")
+tagQueryAttrsHas <- function(els, attrs) {
+  attrs <- as_character2(attrs)
+  if ((length(attrs) == 0) || (!is.character(attrs))) {
+    stop("`attrs` must be a character vector", call. = FALSE)
   }
   unlist(
     tagQueryLapply(els, function(el) {
       if (!isTagEnv(el)) return(FALSE)
-      tagHasAttribute(el, attr)
+
+      for (attr in attrs) {
+        if (!tagHasAttribute(el, attr)) {
+          return(FALSE)
+        }
+      }
+      # All attrs found
+      return(TRUE)
     }),
     use.names = FALSE
   )
 }
 
-
-getCssClass <- function(class) {
+prepCssClass <- function(class) {
   class <- as_character2(class)
   if (length(class) == 0 || !is.character(class)) {
     stop("`class` must resolve to a character value with a length of at least 1")
   }
-  splitCssClass(class)
+  class
+}
+getCssClass <- function(class) {
+  splitCssClass(prepCssClass(class))
 }
 splitCssClass <- function(class) {
   if (length(class) > 1) {
```

Comment:
Should the `strsplit()` below here be `strsplit(class, "\\s+")[[1]]`?

Reply:
41b9ee5